test/e2e/targets/awss3/main.go

```diff
@@ -0,0 +1,364 @@
+					By("sending an event", func() {
+						sentEvent = e2ece.NewHelloEvent(f)
+
+						job := e2ece.RunEventSender(f.KubeClient, ns, trgtURL.String(), sentEvent)
+						apps.WaitForCompletion(f.KubeClient, job)
+					})
+				})
+
+				It("creates an object onto the bucket", func() {
+					var receivedObj []byte
+					var err error
+
+					By("polling the bucket objects", func() {
+						receivedObjs := e2es3.GetObjects(s3Client, bucketName)
+						Expect(receivedObjs).To(HaveLen(1),
+							"Received %d objects instead of 1", len(receivedObjs))
+
+						object := *receivedObjs[0]
+						receivedObj, err = ioutil.ReadAll(object.Body)
+						Expect(err).ToNot(HaveOccurred())
+					})
+
+					By("inspecting the object payload", func() {
+						ObjData := make(map[string]interface{})
+						err := json.Unmarshal(receivedObj, &ObjData)
+						Expect(err).ToNot(HaveOccurred())
+
+						eventData, err := json.Marshal(ObjData)
+						Expect(err).ToNot(HaveOccurred())
+
+						gotEvent := &cloudevents.Event{}
+						err = json.Unmarshal(eventData, gotEvent)
+						Expect(err).ToNot(HaveOccurred())
+
+						Expect(gotEvent.ID()).To(Equal(sentEvent.ID()))
+						Expect(gotEvent.Type()).To(Equal(sentEvent.Type()))
+						Expect(gotEvent.Source()).To(Equal(sentEvent.Source()))
+						Expect(gotEvent.Data()).To(Equal(sentEvent.Data()))
+						Expect(gotEvent.Extensions()[e2ece.E2ECeExtension]).
+							To(Equal(sentEvent.Extensions()[e2ece.E2ECeExtension]))
+					})
+				})
+			})
+		})
+
+		When("the CloudEvent context is discarded", func() {
+			BeforeEach(func() {
+				By("creating an AWSS3Target object with discardCEContext enabled", func() {
+					trgt, err := createTarget(trgtClient, ns, "test-",
+						withARN(bucketARN),
+						withCredentials(awsSecret.Name),
+						withDiscardCEContext(),
+					)
+					Expect(err).ToNot(HaveOccurred())
+
+					trgt = ducktypes.WaitUntilReady(f.DynamicClient, trgt)
+
+					trgtURL = ducktypes.Address(trgt)
+					Expect(trgtURL).ToNot(BeNil())
+				})
+			})
+			When("an event is sent to the target", func() {
+				var sentEvent *cloudevents.Event
+
+				BeforeEach(func() {
+					By("sending an event", func() {
+						sentEvent = e2ece.NewHelloEvent(f)
+
+						job := e2ece.RunEventSender(f.KubeClient, ns, trgtURL.String(), sentEvent)
+						apps.WaitForCompletion(f.KubeClient, job)
+					})
+				})
+
+				It("only puts the event's data onto the bucket object", func() {
+					var receivedObj []byte
+					var err error
+
+					By("polling the bucket objects", func() {
+						receivedObjs := e2es3.GetObjects(s3Client, bucketName)
+						Expect(receivedObjs).To(HaveLen(1),
+							"Received %d objects instead of 1", len(receivedObjs))
+
+						object := *receivedObjs[0]
+						receivedObj, err = ioutil.ReadAll(object.Body)
+						Expect(err).ToNot(HaveOccurred())
+					})
+					By("inspecting the object payload", func() {
+						Expect(receivedObj).To(Equal(sentEvent.Data()))
+					})
+				})
+			})
+		})
+	})
```

Comment:
There is a third mode of operation for the s3 target where an event of type `io.triggermesh.awss3.object.put` can also put data into a bucket without the CE metadata.

Reply:
@cab105 Can you describe a bit more this operation? Not sure If I'm understanding

Reply:
(answering because Chris is probably not up yet 😴)

When `discardCEContext` is `false` (the default), a sender may still indicate that a specific event should be trimmed of its CloudEvent context by sending an event with `event.SetType("io.triggermesh.awss3.object.put")`.

E.g. imagine the following event:

```
Context
  type: io.triggermesh.awss3.object.put
  source: myapp
  id: 00000
Data
  {"orderId": 432748, "item": "Nicolas Cage pillow case"}
```

Only the `Data` part will be stored, regardless of the value of `io.triggermesh.awss3.object.put`, because the event is using the "magic" event type.

Reply:
Gotcha! thanks Antoine!

Reply:
Think @antoineco meant that the `Data` part will be stored regardless of the state of `discardCEContext` when the `io.triggermesh.awss3.object.put` event type is used.